src/figureprinter/FigurePainter.java

```diff
@@ -6,107 +6,173 @@
     }
+
     /**
      * Draws the triangle by specified length number
      * for example when n = 4 , it must be such as
      * drawn below :
+     * <p>
+     * * * * *
+     * * * *
+     * * *
+     * *
      *
-     *       * * * *
-     *       * * *
-     *       * *
-     *       *
-     *
-     * @param length  specified length of triangle will be drawn
+     * @param length specified length of triangle will be drawn
      */
-    public static void drawLeftBottomTriangle(int length){
-        // TODO : complete the method
+    public static void drawLeftBottomTriangle(int length) {
+        for (int i = length; i > 0; i--) {
+            for (int j = 0; j < i; j++) {
+                System.out.print("* ");
+            }
+            System.out.println();
+        }
+        System.out.println("_______________________");
     }
 
     /**
      * Draws the triangle by specified length number
      * for example when n = 4 , it must be such as
      * drawn below :
+     * <p>
+     * * * * *
+     * * * *
+     * * *
+     * *
      *
-     *       * * * *
-     *         * * *
-     *           * *
-     *             *
-     *
-     * @param length  specified length of triangle will be drawn
+     * @param length specified length of triangle will be drawn
      */
-    public static void drawRightBottomTriangle(int length){
-        // TODO : complete the method
+    public static void drawRightBottomTriangle(int length) {
+        for (int i = 0; i < length; i++) {
+            for (int j = 0; j < i; j++) {
+                System.out.print("  ");
+            }
+            for (int j = 0; j < (length - i); j++) {
+                System.out.print("* ");
+            }
+            System.out.println();
+        }
+        System.out.println("_______________________");
     }
+
     /**
      * Draws the rhombus by specified length number
      * for example when n = 5 , it must be such as
      * drawn below :
+     * <p>
+     * *
+     * * *
+     * * * *
+     * * * * *
+     * * * *
+     * * *
+     * *
      *
-     *        *
-     *       * *
-     *      * * *
-     *     * * * *
-     *      * * *
-     *       * *
-     *        *
-     * @param length  specified length of rhombus will be drawn
+     * @param length specified length of rhombus will be drawn
      */
     public static void drawRhombus(int length) {
-        // TODO : complete the method
+        for (int i = 0; i < length - 1; i++) {
+            for (int j = 0; j < (length - i - 1); j++) {
+                System.out.print("  ");
+            }
+            for (int j = 0; j < (i * 2 + 1); j++) {
+                System.out.print("* ");
+            }
+            System.out.println();
+        }
+        for (int i = 0; i < length; i++) {
+            for (int j = 0; j < i; j++) {
+                System.out.print("  ");
+            }
+            for (int j = 0; j < (length * 2 - i * 2 - 1); j++) {
```

Comment:
**j < (length * 2 - i * 2 - 1)** - you don't need this, just put **j < length - i**

Reply:
the same is here